### PR TITLE
mavlink temporary workarounds for dronekit

### PR DESCRIPTION
**Pull request by RomanBapst**
Temporary workaround for companion systems which rely on dronekit.
See commit message for further explanation.
@sanderux Could you review and also correct me if I haven't explained the motivation well enough?

### CODE REVIEW

File: src/modules/mavlink/mavlink_params.c [hunk 1]
```diff
@@ -141,3 +141,27 @@ PARAM_DEFINE_INT32(MAV_FWDEXTSP, 1);
...
  */
 PARAM_DEFINE_INT32(MAV_BROADCAST, 0);
+
+/**
+ * Disable parameter hash check.
+ *
+ * Disabling the parameter hash check functionality will make the mavlink instance
+ * stream parameters continuously.
+ *
+ * @value 0 Hash check enabled.
```
Comment (dagar) [inline src/modules/mavlink/mavlink_params.c]: `@boolean`

File: src/modules/mavlink/mavlink_params.c [hunk 2]
```diff
@@ -141,3 +141,27 @@ PARAM_DEFINE_INT32(MAV_FWDEXTSP, 1);
...
+ */
+PARAM_DEFINE_INT32(MAV_DIS_HASH_CHK, 0);
+
+/**
+ * Disable hearbeat message forwarding.
+ *
+ * The mavlink hearbeat message will not be forwarded if this parameter is set to 'disabled'.
+ * The main reason for disabling heartbeats to be forwarded is because they confuse dronekit.
+ *
+ * @value 0 Forwarding enabled.
```
Comment (dagar) [inline src/modules/mavlink/mavlink_params.c]: `@boolean`

**Comment by dagar** [comment]
Overall looks fine, although you might want to consider using the Param wrapper for this kind of thing. It's fewer moving pieces and provides static checks. https://github.com/PX4/Firmware/blob/master/src/platforms/px4_param.h

File: src/modules/mavlink/mavlink_params.c (hunk 1 above)
Comment (dagar) [reply src/modules/mavlink/mavlink_params.c]: Actually, that probably only makes sense if you want to invert the logic to be parameters that enable.

**Comment by RomanBapst** [comment]
@dagar I addressed your comments except the one regarding the parameter wrapper. Are you suggesting to change the mavlink module to use ModuleParams? If this is what you want then I can do it for all the others parameters as well not just for the ones I added.

**Comment by dagar** [comment]
Yes I'm suggesting to change to ModuleParams (when convenient).

**Comment by RomanBapst** [comment]
@dagar Ok, will do that then.

**Comment by RomanBapst** [comment]
Closing in favor of #10422

**Pull request closed without merging**